xcm/xcm-executor/src/assets.rs

```diff
@@ -140,22 +153,78 @@ impl Assets {
 	/// Return the assets in `self`, but (asset-wise) of no greater value than `assets`.
 	///
 	/// Result is undefined if `assets` includes elements which match to the same asset more than once.
```

Comment:
Can I suggest you to move or add a doc test here? I think it would be a great aid for understanding what this fn supposed to do.